internal/pal/intialize.go

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package pal
+
+func init() {
+	Initialize(func(err error) {
+		if err != nil {
+			panic(err)
+		}
+	})
+}
+
+func Initialize(handler func(err error)) {
```

Comment:
>handler func(err error) [](http://example.com/codeflow?start=16&length=23)

would it be a bit cleaner to define an interface or type for this handler func instead of using raw `func(err error)` everywhere?

Reply:
I've added a more general issue to cover this area of style guidelines.   I think I agree with you (this is what I'd do in c#), but would using interface encourage tighter coupling.  Is the idomatic golang apporach for the caller to define the interface.

Reply:
[https://github.com/microsoft/go-sqlcmd/issues/171](https://github.com/microsoft/go-sqlcmd/issues/171)